src/components/TalentPage/TalentPageLayout.tsx

```diff
@@ -0,0 +1,229 @@
+import { ISkill } from 'components/jobPost/interfaces';
+import { skillsMock } from 'components/jobPost/dataChanges';
+import { Filter, ICategoryBE, SearchSubmitForm } from './interfaces';
+import Select from 'react-select';
+import { selection } from 'components/jobPost/dataChanges';
+import { BsArrowLeftCircle } from 'react-icons/bs';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import MyHiresCompany from './myhires/MyHiresCompany';
+import MySavedTalent from './mysaved/MySavedTalent';
+import FilterProfileUser from './FilterProfileUser';
+import { useGetFilterProfileQuery } from 'service/httpService';
+import Pagination from './Pagination';
+
+const TalentPageLayout: FC = () => {
+	const { t } = useTranslation();
+	const [skillsOption, setSkillsOptions] = useState<ISkill[]>(skillsMock);
+	const [select, setSelect] = useState<ICategoryBE>();
+	const [search, setSearch] = useState<string>('');
+	const [showFilterList, setShowFilterList] = useState<boolean>(true);
+	const [active, setActive] = useState<{ [name: string]: string }>({ ['discover']: 'discover' });
+	const [currentPage, setCurrentPage] = useState<number>(1);
+	const discover = 'discover';
+	const hires = 'hires';
+	const saved = 'save';
+	const paginate = (pageNumber: React.SetStateAction<number>) => setCurrentPage(pageNumber);
+	const {
+		control,
+		formState: { errors },
+	} = useForm<SearchSubmitForm>();
+
+	const filteredSkills = useMemo(() => skillsOption.filter(s => s.value), [skillsOption]);
+	const userSkills = useMemo(() => filteredSkills.map(s => s.name), [filteredSkills]);
+
+	const onSkillsChange = (index: number) => {
+		setSkillsOptions(prevState => {
+			return prevState.map((e, i) => {
+				if (index === i) {
+					return { ...e, value: !e.value };
+				}
+				return e;
+			});
+		});
+	};
+
+	const optionButtons = useMemo(() => {
+		return skillsOption.map((e, i) => (
+			<Label key={e.name} className={`btn btn-${e.value ? 'primary' : 'light'}`}>
+				<input
+					type="checkbox"
+					checked={e.value}
+					autoComplete="off"
+					onChange={() => onSkillsChange(i)}
+				/>
+				{e.name}
+			</Label>
+		));
+	}, [skillsOption]);
+
+	const sendFilter = {
+		select: select?.name,
+		skills: userSkills.toString(),
+		search: search,
+		page: currentPage,
+	};
+	const { data, isLoading } = useGetFilterProfileQuery(sendFilter);
+
+	function getFilterList() {
+		setShowFilterList(!showFilterList);
+	}
+	function useMediaQuery(query: string, defaultMatches = window.matchMedia(query).matches) {
+		const [matches, setMatches] = useState(window.matchMedia(defaultMatches.toString()).matches);
+
+		useEffect(() => {
+			const media = window.matchMedia(query);
+
+			if (media.matches !== matches) setMatches(media.matches);
+
+			const listener = () => setMatches(media.matches);
+
+			media.addEventListener('change', listener);
+
+			return () => media.removeEventListener('change', listener);
+		}, [query, matches]);
+		return matches;
+	}
+
```

Comment:
its mess, I have wrote the order of functions hooks, then handlers then render

Reply:
fixed